Snakefile

```diff
@@ -8,6 +8,69 @@ wildcard_constraints:
 SEGMENTS = ["pb2", "pb1", "pa", "ha","np", "na", "mp", "ns"]
 #SUBTYPES = ["h5n1", "h5nx", "h7n9", "h9n2"]
 
+CURRENT_BASEDIR = workflow.current_basedir # TODO XXX store this value here - can't access within functions because workflow.included_stack is empty
+
+# Load the base config.yaml relative to the entry snakefile (i.e. not this snakefile)
+if os.path.exists(os.path.join(workflow.basedir, 'config.yaml')):
+    configfile: os.path.join(workflow.basedir, 'config.yaml')
+
+# load a config.yaml file if it exists in the current working directory
+if os.path.exists("config.yaml"):
+    configfile: "config.yaml"
+
+from pprint import pp; pp(config, stream=sys.stderr) # TODO XXX remove
+
+class InvalidConfigError(Exception):
+    pass
+
+def resolve_config_path(original_path, wildcards=None):
```

Comment:
Reminder to me: need to resolve any custom rules using this (?) function, currently they're just
```
for rule_file in config.get('custom_rules', []):
    include: rule_file
```

Reply:
(See https://github.com/nextstrain/measles/blob/6584e68622dd305592b98d8630fa94e191b4f164/ingest/Snakefile#L53-L59)

Reply:
There's a decision for us to make here which wasn't needed in measles. The h5n1-cattle-outbreak config uses a custom rule which needs to be searched for within the avian-flu repo. Other workflows use a similar approach for CI - e.g. [zika](https://github.com/nextstrain/zika/blob/649e495a6b181c004bacae26ea4507fc6989abc5/phylogenetic/build-configs/ci/config.yaml#L2). 

Approach 1: Use the same resolving approach as `resolve_config_path` to search for rulefiles. 

Approach 2: Use the approach of measles (custom rules must be relative to the working directory) and replace our usage of `custom_rules` with a "include" directive within `h5n1-cattle-outbreak/Snakefile`. 

There are pros and cons to each I think. Given that we use `custom_rules` in CI, approach 1 is nice as it'll allow CI to be run in a separate directory as desired. Approach 2 also has the "gotcha" where you develop a workflow with custom-rules within `avian-flu`, it all works, but then it won't work when you run it in another directory. There's also the bigger picture issue where if multiple configfiles define `custom_rules` only one of them will be used because they're lists.

Given the above I think using the search order from Approach 1 is the best, but we should also reference `cattle-flu.smk` within `h5n1-cattle-outbreak/Snakefile` so that overlays with `custom_rules` don't have to specify it.

Reply:
Hmm. It feels strange to be letting "custom" rule files be sourced from the workflow source, i.e. "stock custom rules", which seems contradictory.

This is why, FWIW, in measles I intentionally only resolved custom rules files based on the working analysis directory.

For the current "build-configs" pattern like CI or our own automation, I'd think to convert those to analysis directories. I didn't demo this for measles to keep changes to a minimum, but it's an easy change and keeps our interfaces consistent.

Reply:
>  It feels strange to be letting "custom" rule files be sourced from the workflow source, i.e. "stock custom rules", which seems contradictory

Yeah I see that. I guess a clarifying question is "do we want to allow all workflows to be run from an external analysis directory". If that's the case, and I think it probably is, then we either avoid `custom_rules` entirely, expand the interface to have something like `workflow_rules`, or shift our usage of custom rules to their own (sub-)workflows ([concrete example](https://github.com/nextstrain/avian-flu/pull/103/commits/9aeb7b811eb0a3c5f339f40348fb02ecb48b0167)). Is the latter what you mean by "I'd think to convert those to analysis directories"?

If we use different workflows for CI rather than "custom_rules" as we do now, then a cattle-outbreak CI might look like `avian-flu/phylogenetic/ci/h5n1-cattle-outbreak/Snakefile` which would `include: ../../h5n1-cattle-outbreak/Snakefile` which would `include: "../Snakefile"`. I haven't really though through how inheritance like this will work in practice. 


cc @joverlee521

Reply:
To revisit my clarifying question, "do we want to allow all workflows to be run from an external analysis directory", we're answering yes, but saying that CI isn't a "workflow" in and of itself so it's ok that it can't be run from an external analysis directory? I'm not involved much in the CI space, so I'll defer to others here, but on the surface this seems all fine to me.

Reply:
> we're answering yes, but saying that CI isn't a "workflow" in and of itself so it's ok that it can't be run from an external analysis directory? 

I think we have to cut off at some point. The CI seems like a very Nextstrain/workflow developer specific config that would not need to be run by an external user. If someone _really_ wanted to, they can copy the contents of the CI directory to their own analysis directory.

Reply:
> If someone _really_ wanted to

Just for the absence of doubt, here I'm saying that someone is _us_ as it relates to how we run CI (on GitHub actions). Currently we check out the repo and run with `--configfile <ci.yaml>`. With these changes we would be doing the same, but with a slightly different invocation syntax, e.g. as Tom sketched out above. But we wouldn't be using an approach where we ran CI in a separate analysis directory with `nextstrain run ...` as if it were just another workflow. As I said, I'm not involved much in the CI space, but just wanted to explain myself fully here.

Reply:
Ah gotcha. Yeah, for CI I'd think we'd want to test the source code so it'd be running `nextstrain build`.

Reply:
Thanks @joverlee521! 

I've updated the PR to search for custom rules relative to the analysis (working) directory. The `h5n1-cattle-outbreak/Snakemake` workflow now imports it's own "custom" rules as expected. 

I haven't moved the "base" Snakefile to `rules/base.smk`, but this PR is probably the place to do it if we want to.

Reply:
> @jameshadfield: But we wouldn't be using an approach where we ran CI in a separate analysis directory with `nextstrain run ...` as if it were just another workflow.
>
> @joverlee521: Yeah, for CI I'd think we'd want to test the source code so it'd be running `nextstrain build`.

We could have this either way. It's very doable to use `nextstrain run` for this, and we might want to do so in order for it to be an instructive example/regularly test that bit of interface.